pkg/pillar/pubsub/pubsub.go

```diff
@@ -63,6 +63,12 @@ const (
 	fixedName = "zededa"
 	fixedDir  = "/var/tmp/" + fixedName
 	maxsize   = 65535 // Max size for json which can be read or written
+
+	// Copied from types package to avoid loop
+	// PersistDir - Location to store persistent files.
+	PersistDir = "/persist"
+	// PersistConfigDir is where we keep some configuration across reboots
+	PersistConfigDir = PersistDir + "/config"
```

Comment:
Hmm.. Types depending on PubSub seems a bit odd.. Is it because of the EnsureGC file? 
Wondering of we are better off moving EnsureGC to some Utils - Utils can depend on both PubSub and Types..

Reply:
Let me look into a utils pkg.

Reply:
Note that I don't think it makes sense to have pubsub depend on types, since pusub is supposed to be generic.
Should I move all of locationconsts.go from the types package to the utils package? That would be easy but change lots of lines of code.

Reply:
Hmm.. locationconst.go is for the rest of Pillar components which depend on types.
If the hierarchy is -- Types depends on Pubsub -- Then we have two choices:

1) Take the locations as a parameter when we create the pubsub - This avoids defining a constant in Pubsub
2) What you have done - duplicate the definitions - but with the understanding that these definitions are semantically different from those in locationconst.go
      - If some service in pillar needs to refer to pubsub locations, they should use pubsub.<const>

I prefer (1) - That makes PubSub a very generic package..